### PR TITLE
Grape vine development

### PR DESCRIPTION
**Pull request by HamishBrownPFR**
working on #1822

### CODE REVIEW

File: Models/MicroClimate/MicroClimate.cs [hunk 1]
```diff
@@ -298,15 +298,15 @@ private void CalculateStripZoneShortWaveRadiation()
...
 
                 if ((c.Canopy.Height - c.Canopy.Depth) > 0 && c.Canopy.Width <= Wt)
-                    TallestIsVine = true;
+                     TallestIsVine = true;
             }
             if (TallestIsTree)
                 DoTreeRowCropShortWaveRadiation(ref tallest, ref shortest);
-            if (TallestIsVine)
+            else if (TallestIsVine)
                 DoVineStripShortWaveRadiation(ref tallest, ref shortest);
```
Comment (hol353) [inline Models/MicroClimate/MicroClimate.cs:310]: This highlights that we need to refactor MicroClimate at some point. I don't like if statements that have specific sections for whether a tree is the tallest or the vine is the tallest. There has to be a better way. A job for the future.

**Comment by hol353** [comment]
Is this really grape vine? There is an orchid apsimx file in this PR.

**Comment by junqi108** [comment]
> Is this really grape vine? There is an orchid apsimx file in this PR.

yeah, the grapevine model needs this urgently, as the error makes the transpiration for the leaf canopy too low.

**Comment by HamishBrownPFR** [comment]
The Orchard model has a "GrapeVine" cultivar in it and I have fixed some things in there to make sure both Grapevine an the SimpleFruitTree "GrapeVine" cultivar are working well.

File: Models/MicroClimate/MicroClimate.cs (hunk 1 above)
Comment (HamishBrownPFR) [reply Models/MicroClimate/MicroClimate.cs:310]: Yeah, we are aware of this one.   It should be possible to harmonize the three methods code so a single method works well for all three situations.  Currently the StripCrop model works for crops where there is no horizontal overlap, the TreeRowCrop works for situations where there is horizontal overlap and the VineRowCrop works for situations where the tall crop does not fully cover its own row.   If not we could have some component we drop into the simulation to select which strip crop model is used.

**Pull request merged**